services/uploads-scan/serverless.yml

```diff
@@ -0,0 +1,186 @@
+# Welcome to serverless. Read the docs
+# https://serverless.com/framework/docs/
+
+# Serverless.yml is the configuration the CLI
+# uses to deploy your code to your provider of choice
+
+# The `service` block is the name of the service
+service: uploads-scan
+
+# The `provider` block defines where your service will be deployed
+provider:
+  name: aws
+  runtime: nodejs12.x
+  region: us-east-1
+  stage: dev
+
+plugins:
+  - serverless-plugin-scripts
+  - serverless-s3-remover
+
+custom:
+  stage: ${opt:stage, self:provider.stage}
+  attachments_bucket_arn: ${cf:uploads-${self:custom.stage}.AttachmentsBucketArn}
+  attachments_bucket_name: ${cf:uploads-${self:custom.stage}.AttachmentsBucketName}
+
+  scripts:
+    hooks:
+      # This script is run locally when running 'serverless deploy'
+      package:initialize: |
+        set -e
+        curl -L --output lambda_layer.zip https://github.com/sethsacher/lambda-clamav-layer/releases/download/0.5/lambda_layer.zip
+      deploy:finalize: |
+        rm lambda_layer.zip
+
+  remover:
+     buckets:
+       - uploads-scan-${opt:stage, self:provider.stage}-avscan
+
+layers:
+  clamDefs:
+    package:
+      artifact: lambda_layer.zip
+
+functions: 
+  avScan:
+    handler: antivirus.lambdaHandleEvent
+    name: uploads-scan-${opt:stage, self:provider.stage}-avScan
+    events:
+      - s3: 
+          bucket: ${self:custom.attachments_bucket_name}
+          event: s3:ObjectCreated:* # Triggers Lambda once per file, not once per upload
+          existing: true
+    role: BucketAVScanRole
+    timeout: 300 # 300 seconds = 5 minutes. Average scan is 25 seconds.
+    memorySize: 3008 # Max lambda memory allocation; clamscan didn't work with 1024
+    layers:
+      - { Ref: ClamDefsLambdaLayer }
+    environment: 
+      CLAMAV_BUCKET_NAME: !Ref AVBucket
+      PATH_TO_AV_DEFINITIONS: 'lambda/s3-antivirus/av-definitions'
+  avDownloadDefinitions:
+    handler: download-definitions.lambdaHandleEvent
+    events:
+      - schedule: cron(0 */6 * * ? *)
+    role: BucketAVDownloadRole
+    timeout: 300 # 300 seconds = 5 minutes
+    memorySize: 1024
+    layers:
+      - { Ref: ClamDefsLambdaLayer }
+    environment: 
+      CLAMAV_BUCKET_NAME: !Ref AVBucket
+      PATH_TO_AV_DEFINITIONS: 'lambda/s3-antivirus/av-definitions'
+
+resources:
+  Resources:
+    AVBucket:
+      Type: AWS::S3::Bucket
+      Properties: 
+        BucketName: uploads-scan-${opt:stage, self:provider.stage}-avscan
+        AccessControl: PublicRead
```

Comment:
This is the clamav definitions bucket, right?  I don't think this is required.  Looking at the code, the role's for the lambda's include permissions to read/write to this bucket and it's objects.  
Can probably set this to Private

Reply:
Right now, the AVScan Lambda only has access granted to the Attachments Bucket and the AVDownloadDefinitions Lambda has access to the Definitions Bucket. I can change this to Private and assign permissions to the AVScan Lambda, but the contents didn't seem sensitive.